[ADD] procurement_jit_assign_move

gurneyalex opened the pull request:
This module changes the behavior of procurement_jit in 8.0 to avoid having
manually check availability on all the stock pickings.

See https://github.com/odoo/odoo/issues/7773 for details

gurneyalex committed:
[ADD] procurement_jit_assign_move

This module changes the behavior of procurement_jit in 8.0 to avoid having
manually check availability on all the stock pickings.

See https://github.com/odoo/odoo/issues/7773 for details

gurneyalex commented:
@alexis-via @jco-odoo your opinion on this PR is welcome.

guewen commented on procurement_jit_assign_move/tests/test_procument.py:
```diff
@@ -0,0 +1,21 @@
+class ProcurementTest(TransactionCase):
+
+    def test_something(self):
+        values = {'company_id': 1,
```
`self.env.ref('base.main_company').id`

guewen commented on procurement_jit_assign_move/tests/test_procument.py:
```diff
@@ -0,0 +1,21 @@
+
+class ProcurementTest(TransactionCase):
+
+    def test_something(self):
```
A better name? :-)

lepistone replied:
Well, the name is not wrong as such, but...

lepistone commented on procurement_jit_assign_move/README.rst:
```diff
@@ -0,0 +1,62 @@
+This module extends the functionality of procurement_jit to assign moves for
+make to stock procurements. 
+
+This compensate the change of features reported in
```
s

lepistone commented:
:+1:

tafaRU commented on procurement_jit_assign_move/README.rst:
```diff
@@ -0,0 +1,62 @@
+check there if your issue has already been reported. If you spotted it first,
+help us smashing it by providing a detailed and welcomed `feedback
+<https://github.com/OCA/
+{project_repo}/issues/new?body=module:%20
```
Replace placeholder by `stock-logistics-workflow`

tafaRU commented on procurement_jit_assign_move/README.rst:
```diff
@@ -0,0 +1,62 @@
+help us smashing it by providing a detailed and welcomed `feedback
+<https://github.com/OCA/
+{project_repo}/issues/new?body=module:%20
+stock-logistics-workflow%0Aversion:%20
```
Replace it by the module name ;-)

guewen commented:
:+1:

lmignon commented:
:+1:  (code review + functional tests)

tafaRU commented:
@gurneyalex, few nitpicking comments otherwise :+1:

gurneyalex commented:
thanks for the reviews

jco-odoo commented on procurement_jit_assign_move/models/stock_move.py:
```diff
@@ -0,0 +1,19 @@
+        res = super(StockMove, self).action_confirm()
+        make_to_stock_moves = self.filtered(
+            lambda x:
+            x.procurement_id.rule_id.procure_method == 'make_to_stock'
```
Why not use x.procure_method immediately?  It is different from the procurement_jit in 9 as that one only does it for sale orders, not for just any stock move.  I guess it depends on what you want to achieve.

gurneyalex replied:
@jco-odoo for no good reason... I'd totally forgotten that field on the move. :blush: 

Thanks for your feedback.

tafaRU commented:
@gurneyalex, Bug Tracker link is still wrong, please see https://github.com/gurneyalex/stock-logistics-workflow/pull/2 to fix it.

gurneyalex commented:
@tafaRU fixed thanks

tafaRU commented:
@gurneyalex and what about https://github.com/OCA/stock-logistics-workflow/pull/183#discussion_r46297097 ?

gurneyalex committed:
split the tests

procurement_jit_assign_move pulls in procurement_jit_stock
which can break the tests of other modules in the project

gurneyalex commented:
ok globally green except for a tinyweeny coveralls issue. I think this can be merged.

alexis-via commented:
:+1: 
Let's merge !
